tests/components/unifi/test_button.py

```diff
@@ -121,33 +121,44 @@ def mock_secret():
 ]
 
 
-async def _test_button_entity(
+@pytest.mark.parametrize("device_payload", [DEVICE_RESTART + DEVICE_POWER_CYCLE_POE])
+@pytest.mark.parametrize("wlan_payload", [WLAN_REGENERATE_PASSWORD])
+@pytest.mark.parametrize(
+    "site_payload",
+    [
+        [{"desc": "Site name", "name": "site_id", "role": "admin", "_id": "1"}],
+        [{"desc": "Site name", "name": "site_id", "role": "not admin", "_id": "1"}],
+    ],
+)
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_entity_and_device_data(
     hass: HomeAssistant,
     entity_registry: er.EntityRegistry,
+    config_entry_factory,
+    site_payload: dict[str, Any],
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Validate entity and device data with and without admin rights."""
+    with patch("homeassistant.components.unifi.PLATFORMS", [Platform.BUTTON]):
+        config_entry = await config_entry_factory()
+    if site_payload[0]["role"] == "admin":
+        await snapshot_platform(hass, entity_registry, snapshot, config_entry.entry_id)
+    else:
+        assert len(hass.states.async_entity_ids(BUTTON_DOMAIN)) == 0
```

Comment:
Why can't we have a snapshot for this case too? We don't want conditions in tests.

Reply:
The snapshot_platform complains that all entities should be enabled

Reply:
Can we enable all the entities for the test?

https://github.com/home-assistant/core/blob/482cf261c0d7cf356dd0ffcb6c2e2eb0a8dd8bab/tests/components/conftest.py#L44

Reply:
If we just want to test that the entity is or isn't enabled, we shouldn't use snapshots at all.

Reply:
With admin rights they are enabled. And without admin rights they are not. Alternatively id have to make two tests here